static/app/components/hotkeys.tsx

```diff
@@ -0,0 +1,113 @@
+import styled from '@emotion/styled';
+
+import space from 'sentry/styles/space';
+import {getKeyCode} from 'sentry/utils/getKeyCode';
+
+const macModifiers = {
+  18: '⌥',
+  17: '⌃',
+  91: '⌘',
+};
+
+const normalModifiers = {
+  18: 'ALT',
+  17: 'CTRL',
+  91: 'CMD',
+};
+
+const genericGlyphs = {
+  16: '⇧',
+  8: '⌫',
+  37: '←',
+  38: '↑',
+  39: '→',
+  40: '↓',
+  9999: '+',
+};
+
+const keyToDisplay = (
+  key: string,
+  platform?: 'macos' | string
+): [React.ReactNode, boolean] => {
+  let keyStr = key.toUpperCase();
+  let hasMissing = false;
+
+  const keyCode = getKeyCode(key);
+  if (keyCode) {
+    const isMac = platform
+      ? platform === 'macos'
+      : window?.navigator?.platform?.toLowerCase().startsWith('mac') ?? false;
+
+    const modifierMap = isMac ? macModifiers : normalModifiers;
+    const keyStr = modifierMap[keyCode] ?? key.toUpperCase();
+    
+    if (keyStr === 'CMD') {
+      hasMissing = true;
+    }
+
+    const glyph = genericGlyphs[keyCode];
+
+    if (glyph) {
+      keyStr = glyph;
+    }
+  }
+
+  // eslint-disable-next-line react/jsx-key
+  return [<Key>{keyStr}</Key>, hasMissing];
+};
+
+const Hotkeys = ({
+  value,
+  platform,
+}: {
+  /**
+   * Pass key combinations in with + as the separator.
+   * For example: command+option+x
+   *
+   * Use comma for fallback key combos when the first one contains a key that does not exist on that os (non-mac):
+   * command+option+x,ctrl+shift+x
+   * (does not have to be the same combo)
+   *
+   * Escape the + key with a slash |+
+   */
+  value: string;
+  platform?: 'macos' | string;
+}) => {
+  // Split by commas and then split by +, but allow escaped /+
+  const byPlatform = value.split(',').map(o => o.trim().split(/(?<!\\)\+/g));
+
+  let firstSetHasMissing = false;
+  let output = byPlatform[0].map(key => {
+    const [node, hasMissing] = keyToDisplay(key, platform);
+
+    if (hasMissing) {
+      firstSetHasMissing = true;
+    }
+
+    return node;
+  });
```

Comment:
so with my above suggestion this could look more like this

```tsx
const isMac = platform
  ? platform === 'macos'
  : window?.navigator?.platform?.toLowerCase().startsWith('mac') ?? false;

const keys = byPlatform.map(keyToDisplay);

// If we're not using mac find the first key that is generic.
// Otherwise show whatever the first hotkey is. 
const selectedkey = keys.first(key => !isMac ? key.specificToOs === 'generic' : true);

// No key available for the OS. Don't show a hotkey
if (selectedKey === undefined) {
  return null
}

return <HotkeysContainer>{output}</HotkeysContainer>;
```